### PR TITLE
fix(dropdown): rename popup attribute to aria-controls

### PR DESCRIPTION
**Pull request by ashleyryan**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] clarity.design website / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

The popup attribute conflicts with the Popup API that chromium has started implementing and results in errors logged in the console. See development status: https://chromestatus.com/feature/5463833265045504

Issue Number: #140

## What is the new behavior?

This renames the attribute to aria-controls, which is semantically correct. This is a breaking change

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

Applications will lose aria attribute setting if they don't rename the attribute in their code. The component is a beta components so this change should be acceptable.

### CODE REVIEW

File: package-lock.json
Comment (kevinbuhmann) [inline package-lock.json:128474]: Where did this come from? There is no package.json change.

File: projects/core/src/dropdown/dropdown.stories.mdx
Comment (hippee-lee) [inline projects/core/src/dropdown/dropdown.stories.mdx:13]: Breaking changes need a major version bump, this looks like a minor version bump.

File: projects/core/src/internal/controllers/aria-popup-trigger.controller.spec.ts
Comment (hippee-lee) [inline projects/core/src/internal/controllers/aria-popup-trigger.controller.spec.ts:15]: 👍🏼

**Review by hippee-lee — CHANGES_REQUESTED** [review]
This is a breaking change for consumers already using any of the components that use the internal popover. Following semantic versioning it should go into v7 when that is released.

File: projects/core/src/internal/controllers/aria-popup-trigger.controller.ts
Comment (hippee-lee) [inline projects/core/src/internal/controllers/aria-popup-trigger.controller.ts:23]: 👍🏼

File: projects/core/src/dropdown/dropdown.stories.mdx
Comment (ashleyryan) [reply projects/core/src/dropdown/dropdown.stories.mdx:13]: It is a breaking change, but we also clearly spell out that the component is in beta and may break. I would prefer not to hold this til a next major version.

File: projects/core/src/internal/controllers/aria-popup-trigger.controller.ts
Comment (ashleyryan) [reply projects/core/src/internal/controllers/aria-popup-trigger.controller.ts:23]: I'm a little confused about attributes vs properties, especially when the attribute has a dash. I'm not sure if I should put `ariaControls` back on the `AriaPopupTrigger` type and check for either the property or the attribute here?

File: projects/core/src/dropdown/dropdown.stories.mdx
Comment (hippee-lee) [reply projects/core/src/dropdown/dropdown.stories.mdx:13]: It your call, I'm not a maintainer any more :) 
It will break all of my apps tho.

File: projects/core/src/internal/controllers/aria-popup-trigger.controller.ts
Comment (hippee-lee) [inline projects/core/src/internal/controllers/aria-popup-trigger.controller.ts:23]: I think what you changed it to is fine, HTML spec says both should work. I believe that the `aria-controls` attributes on the element are converted to the `ariaControls` property once the browser DOM node object.

**Review by hippee-lee — APPROVED** [review]

**Review by kevinbuhmann — APPROVED** [review]

**Pull request merged**
